### PR TITLE
logits API support

### PR DESCRIPTION
**Pull request by qmeng222**

### CODE REVIEW

**Comment by JoyboyBrian** [comment]
LGTM

**Comment by zhiyuan8** [comment]
Fix merge conflicts and ready to merge

**Comment by qmeng222** [comment]
`git checkout main`
`git pull origin main`
`git checkout qingying-logits`
`git merge main`
Accept current change:

**Comment by zhiyuan8** [comment]
@JoyboyBrian plz check merge conflicts, we just updated llama.cpp

**Review by JoyboyBrian — COMMENTED** [review]
conflicts resolved
